### ISSUE TITLE
Simple Error in v1.3.0 in combination with sapphire.js

### ISSUE THREAD

**Issue by Ronnie1320** [discordjs/discord.js#8602]
### Which package is this bug report for?

discord.js

### Issue description

1) Execute `sapphire new` and choose the TS template. 
2) Run `npm run dev` to start the dev server. 
3) Get this Error: 
```
node_modules/discord.js/typings/index.d.ts:1635:75 - error TS2344: Type 'U' does not satisfy the constraint 'ModalActionRowComponentResolvable | MessageActionRowComponentResolvable'.
  Type 'U' is not assignable to type 'Required<BaseMessageComponentOptions> & MessageSelectMenuOptions'.
    Type 'U' is not assignable to type 'Required<BaseMessageComponentOptions>'.

1635   public constructor(data?: MessageActionRow<T> | MessageActionRowOptions<U> | V);
                                                                               ~

  node_modules/discord.js/typings/index.d.ts:1630:3
    1630   U = T extends ModalActionRowComponent ? ModalActionRowComponentResolvable : MessageActionRowComponentResolvable,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    This type parameter might need an `extends Required<BaseMessageComponentOptions>` constraint.
  node_modules/discord.js/typings/index.d.ts:1630:3
    1630   U = T extends ModalActionRowComponent ? ModalActionRowComponentResolvable : MessageActionRowComponentResolvable,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    This type parameter might need an `extends Required<BaseMessageComponentOptions> & MessageSelectMenuOptions` constraint.
  node_modules/discord.js/typings/index.d.ts:1630:3
    1630   U = T extends ModalActionRowComponent ? ModalActionRowComponentResolvable : MessageActionRowComponentResolvable,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    This type parameter might need an `extends ModalActionRowComponentResolvable | MessageActionRowComponentResolvable` constraint.


Found 1 error in node_modules/discord.js/typings/index.d.ts:1635

ERROR: "build" exited with 2.
```

If you find the File and the Line, you can simply autofix it with VS Code. 

Upgrading to 13.0.3 doesn´t help, and 14.0+ is not supported by Sapphire.js

### Package version

13.0.2

### Node.js version

17.7.1

### Operating system

Windows
Docker doesn´t work too

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

No Intents

**Comment by imranbarbhuiya** [comment]
this issue occurs with ts v4.8.2
it's fixed in https://github.com/discordjs/discord.js/pull/8601

**Comment by Ronnie1320** [comment]
So what should I change? Updating TS to 4.8 doesn´t help, also not updateing discord-api-types.

**Comment by Ronnie1320** [comment]
THX, I'll try it. 
Just now, I used a script with sed.

**Comment by Jiralite** [comment]
Resolved with #8601.
